tests/unit/test_serialization.py

```diff
@@ -755,3 +776,42 @@ def test_serialize_deserialize_double_round_trip(deserialize, serialize, seriali
 
             np.testing.assert_array_equal(nan_a, nan_b)
             np.testing.assert_allclose(field_result_a[~nan_a], field_result_b[~nan_b], rtol=1e-15)
+
+
+@pytest.mark.parametrize(
```

Comment:
This test and the helper function might not be needed.
The existing test cases and functions already go through it all. Can it not be included there?

Reply:
Because this would create a lot more parameterized versions of a simple test

Reply:
Test combined in the latest commit.